test/Array.ts

```diff
@@ -339,6 +339,22 @@ describe('Array', () => {
   })
 
   it('uniq', () => {
+    interface A {
+      a: string
+      b: number
+    }
+
+    const setoidA = contramap((f: A) => f.b, ordNumber)
+    const arrA: A = { a: 'a', b: 1 }
+    const arrB: A = { a: 'b', b: 1 }
+    const arrC: A = { a: 'c', b: 2 }
+    const arrD: A = { a: 'd', b: 2 }
+    const arrUniq = [arrA, arrC]
+
+    assert.equal(uniq(setoidA)(arrUniq), arrUniq, 'Preserve original array')
+    assert.deepEqual(uniq(setoidA)([arrA, arrB, arrC, arrD]), [arrA, arrC])
```

Comment:
@alex-ketch  Can you  please ad those for specifically checking relative order ?
```typescript
    assert.deepEqual(uniq(fooSeoid)([fooA, fooC]), [fooA, fooC])
    assert.deepEqual(uniq(fooSeoid)([fooC, fooA]), [fooC, fooA])
```